frame/elections-phragmen/src/lib.rs

```diff
@@ -627,36 +510,206 @@ decl_module! {
 		/// # <weight>
 		/// The total number of voters and those that are defunct must be provided as witness data.
 		/// # </weight>
-		#[weight = T::WeightInfo::clean_defunct_voters(*_num_voters, *_num_defunct)]
-		fn clean_defunct_voters(origin, _num_voters: u32, _num_defunct: u32) {
+		#[pallet::weight(T::WeightInfo::clean_defunct_voters(*_num_voters, *_num_defunct))]
+		pub(crate) fn clean_defunct_voters(
+			origin: OriginFor<T>,
+			_num_voters: u32,
+			_num_defunct: u32,
+		) -> DispatchResultWithPostInfo {
 			let _ = ensure_root(origin)?;
 			<Voting<T>>::iter()
 				.filter(|(_, x)| Self::is_defunct_voter(&x.votes))
-				.for_each(|(dv, _)| {
-					Self::do_remove_voter(&dv)
-				})
+				.for_each(|(dv, _)| Self::do_remove_voter(&dv));
+
+			Ok(None.into())
 		}
+	}
 
-		/// What to do at the end of each block.
+	#[pallet::event]
+	#[pallet::metadata(<T as frame_system::Config>::AccountId = "AccountId", BalanceOf<T> = "Balance")]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A new term with \[new_members\]. This indicates that enough candidates existed to run
+		/// the election, not that enough have has been elected. The inner value must be examined
+		/// for this purpose. A `NewTerm(\[\])` indicates that some candidates got their bond
+		/// slashed and none were elected, whilst `EmptyTerm` means that no candidates existed to
+		/// begin with.
+		NewTerm(Vec<(<T as frame_system::Config>::AccountId, BalanceOf<T>)>),
```

Comment:
if you want to have a proper metadata for this one you also need to add:
`Vec<(<T as frame_system::Config>::AccountId, BalanceOf<T>)> = "Vec<(AccountId, Balance)>"`

Maybe we can improve pallet macro here so that type is change automatically for this, not sure how difficult it is, and should be solved by scale-info anyway.

Reply:
Good point, fixed.